plugins/inputs/redis/redis.go

```diff
@@ -352,6 +463,126 @@ func gatherInfoOutput(
+	fields["migrate_cached_sockets"] = o.MigrateCachedSockets
+	fields["module_fork_in_progress"] = o.ModuleForkInProgress
+	fields["module_fork_last_cow_size"] = o.ModuleForkLastCowSize
+	fields["number_of_cached_scripts"] = o.NumberOfCachedScripts
+	fields["pubsub_channels"] = o.PubsubChannels
+	fields["pubsub_patterns"] = o.PubsubPatterns
+	fields["rdb_bgsave_in_progress"] = o.RdbBgsaveInProgress
+	fields["rdb_changes_since_last_save"] = o.RdbChangesSinceLastSave
+	fields["rdb_current_bgsave_time_sec"] = o.RdbCurrentBgsaveTimeSec
+	fields["rdb_last_bgsave_status"] = o.RdbLastBgsaveStatus
+	fields["rdb_last_bgsave_time_sec"] = o.RdbLastBgsaveTimeSec
+	fields["rdb_last_save_time"] = o.RdbLastSaveTime
+	fields["rdb_last_save_time_elapsed"] = o.RdbLastSaveTimeElapsed
+	fields["rdb_last_cow_size"] = o.RdbLastCowSize
+	fields["redis_version"] = o.RedisVersion
+	fields["rejected_connections"] = o.RejectedConnections
+	fields["repl_backlog_active"] = o.ReplBacklogActive
+	fields["repl_backlog_first_byte_offset"] = o.ReplBacklogFirstByteOffset
+	fields["repl_backlog_histlen"] = o.ReplBacklogHistlen
+	fields["repl_backlog_size"] = o.ReplBacklogSize
+	fields["rss_overhead_bytes"] = o.RssOverheadBytes
+	fields["rss_overhead_ratio"] = o.RssOverheadRatio
+	fields["second_repl_offset"] = o.SecondReplOffset
+	fields["slave_expires_tracked_keys"] = o.SlaveExpiresTrackedKeys
+	fields["sync_full"] = o.SyncFull
+	fields["sync_partial_err"] = o.SyncPartialErr
+	fields["sync_partial_ok"] = o.SyncPartialOk
+	fields["total_commands_processed"] = o.TotalCommandsProcessed
+	fields["total_connections_received"] = o.TotalConnectionsReceived
+	fields["total_net_input_bytes"] = o.TotalNetInputBytes
+	fields["total_net_output_bytes"] = o.TotalNetOutputBytes
+	fields["total_reads_processed"] = o.TotalReadsProcessed
+	fields["total_system_memory"] = o.TotalSystemMemory
+	fields["total_writes_processed"] = o.TotalWritesProcessed
+	fields["tracking_clients"] = o.TrackingClients
+	fields["tracking_total_items"] = o.TrackingTotalItems
+	fields["tracking_total_keys"] = o.TrackingTotalKeys
+	fields["tracking_total_prefixes"] = o.TrackingTotalPrefixes
+	fields["unexpected_error_replies"] = o.UnexpectedErrorReplies
+	fields["uptime"] = o.Uptime
+	fields["used_cpu_sys"] = o.UsedCPUSys
+	fields["used_cpu_sys_children"] = o.UsedCPUSysChildren
+	fields["used_cpu_user"] = o.UsedCPUUser
+	fields["used_cpu_user_children"] = o.UsedCPUUserChildren
+	fields["used_memory"] = o.UsedMemory
+	fields["used_memory_dataset"] = o.UsedMemoryDataset
+	fields["used_memory_dataset_perc"] = o.UsedMemoryDatasetPerc
+	fields["used_memory_lua"] = o.UsedMemoryLua
+	fields["used_memory_overhead"] = o.UsedMemoryOverhead
+	fields["used_memory_peak"] = o.UsedMemoryPeak
+	fields["used_memory_peak_perc"] = o.UsedMemoryPeakPerc
+	fields["used_memory_rss"] = o.UsedMemoryRss
+	fields["used_memory_scripts"] = o.UsedMemoryScripts
+	fields["used_memory_startup"] = o.UsedMemoryStartup
```

Comment:
so one downside of this is that you're going to get all these fields reported in InfluxDB even if Redis didn't report them. One way you could solve this is to only add the field if it also exists in the `fields` map.